e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts

```diff
@@ -1,35 +1,138 @@
-import { expect } from 'vitest';
+import { Tree } from '@nx/devkit';
+import { rm } from 'node:fs/promises';
+import { join } from 'node:path';
+import { afterEach, expect } from 'vitest';
+import { generateCodePushupConfig } from '@code-pushup/nx-plugin';
+import {
+  generateWorkspaceAndProject,
+  materializeTree,
+} from '@code-pushup/test-nx-utils';
 import { removeColorCodes } from '@code-pushup/test-utils';
-import { executeProcess } from '@code-pushup/utils';
+import { distPluginPackage, executeGenerator } from '../mocks/utils';
+
+function executeConfigurationGenerator(
+  args: string[],
+  cwd: string = process.cwd(),
+) {
+  return executeGenerator(args, {
+    bin: distPluginPackage(cwd),
+    generator: 'configuration',
+    cwd,
+  });
+}
 
 describe('nx-plugin g configuration', () => {
-  it('should run configuration generator on react-todos-app', async () => {
-    const { code, stderr, stdout } = await executeProcess({
-      command: 'npx',
-      args: [
-        'nx',
-        'g',
-        '@code-pushup/nx-plugin:configuration react-todos-app --dryRun',
-      ],
-    });
+  let tree: Tree;
+  const project = 'my-lib';
+  const projectRoot = join('libs', project);
+  const baseDir = 'tmp/nx-plugin-e2e/generators/configuration';
+
+  const generatorExecMsgRegex = (cwd: string) =>
+    `NX  Generating ${distPluginPackage(cwd)}:configuration`;
+  const createConfigMsgRegex = /^CREATE.*code-pushup.config.ts/m;
+  const updateProjectMsgRegex = /^UPDATE.*project.json/m;
+
+  beforeEach(async () => {
+    tree = await generateWorkspaceAndProject(project);
+  });
+
+  afterEach(async () => {
+    await rm(baseDir, { recursive: true, force: true });
+  });
+
+  it('should inform about dry run', async () => {
+    const cwd = join(baseDir, 'configure');
+    await materializeTree(tree, cwd);
+
+    const { stderr } = await executeConfigurationGenerator(
```

Comment:
These tests just became way less readable :disappointed: Before it was simple and direct, now everything's extracted to wrapper functions and constants.

For example, previously the test simply ran `executeProcess` with the CLI command, now it's "abstracted" away in a `executeConfigurationGenerator` via another  `executeGenerator` wrapper. This obfuscates the _actual user interaction_ which is being tested.

The _user_ runs the `nx g @code-pushup/nx-plugin:configuration ...` commands and sees the `CREATE ... code-pushup.config.ts` logs, so extracting those things out of the tests is counter-productive. When I read an `it`-block, I want to understand what's happening without having to click through a bunch of testing "abstractions" and extracted setup code. 

It's like instead of writing a simple statement like `console.log('Hello, world!')`, you break it down and spread it around as if to hide how simple it could be :sweat_smile: 

```ts
// pointlessly extracted constant
import { NAME } from './some-other-file-you-have-to-know-about'

// ... code to scroll through ...

// pointless wrapper function
function msg(prefix: string, suffix: string = '!') {
  return `${prefix}${NAME}${suffix}`;
}

// ... more code to scroll through ...

// another pointless wrapper
const log = () => {
  console.log(msg('Hello ', '!'));
}

// ... more code to scroll through ...

// "It's slightly shorter, that means the code is better, right?"
log();
```

Reply:
refactored

Reply:
Thanks for the refactor, now I can understand each test by reading it top to bottom :slightly_smiling_face: